frontend/src/components/profile/PageLayout.jsx

```diff
@@ -10,15 +10,34 @@ import EditButton from './EditButton';
 class PageLayout extends React.Component {
     constructor(props) {
         super(props);
-        this.state = { key: 'aboutMe' }
+        this.state = { key: 'aboutMe', items: [], isLoaded:false }
     }    
 
     handleClick = e => {
         this.setState({key: e.key});
     };    
 
+    componentDidMount() {
+
+        var github_user = (this.props.profile.github.replace("http://github.com/", ''))
+
+        fetch('https://api.github.com/users/' + github_user + '/events')
```

Comment:
this code requires the username so you need to do generic parsing to get the username from a the url

Reply:
`PageLayout.jsx:24 GET https://api.github.com/users/https://github.com/AnkushSharma2698/events` This request is a 404 from github or it returns json that is not a list

Reply:
3 things need to be done to genericize the view. 

1. url parsing to get username
2. error handling on a request that does not succeed
3. Move the code to a diff component